### PR TITLE
Create pull_request_template.md

### PR DESCRIPTION
**Pull request by baiyli**
*Issue #, if available:*
- N/A

*Description of changes:*
- Create pull_request_template.md

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by xiaoxshe — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - [ ] Check the PR title and description for clarity. It should describe the changes made and the reason behind them.
+    - [ ] Ensure that the PR follows the contribution guidelines, if applicable.
+2. Security requirements
+    - [ ] Ensure that a Pull Request (PR) does not expose passwords and other sensitive information by using git-secrets and upload relevant evidence: https://github.com/awslabs/git-secrets
+    - [ ] Ensure commit has GitHub Commit Signature
+3. Automatic checks in GitHub Workflow
+    - [ ] collab-check : Check if the owner of PR belongs is collaborator of this repo or not
+    - [ ] wait-for-approval : Since the owner of PR does not belong to this repo, manual approval from corresponding GitHub team is required
+    - [ ] unit-tests : Run unit tests based on python version from 3.8 to 3.11. If any unit test failed, PR will be blocked here.
+    - [ ] integ-tests : Run integ tests based on python version from 3.8 to 3.11. If any unit test failed, PR will be blocked here.
```
Comment (jswudi) [inline .github/pull_request_template.md:15]: it's hard for them to run integ tests

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+## For Requester
+
+1. Description
+    - [ ] Check the PR title and description for clarity. It should describe the changes made and the reason behind them.
+    - [ ] Ensure that the PR follows the contribution guidelines, if applicable.
+2. Security requirements
+    - [ ] Ensure that a Pull Request (PR) does not expose passwords and other sensitive information by using git-secrets and upload relevant evidence: https://github.com/awslabs/git-secrets
+    - [ ] Ensure commit has GitHub Commit Signature
+3. Automatic checks in GitHub Workflow
+    - [ ] collab-check : Check if the owner of PR belongs is collaborator of this repo or not
```
Comment (jswudi) [inline .github/pull_request_template.md:12]: why PR owner is collaborator? anyone can be collaborator?

File: .github/pull_request_template.md (hunk 1 above)
Comment (baiyli) [reply .github/pull_request_template.md:15]: Yeah, I actually mean that request can check github workflow result. I can remove this part if it is confusing.

File: .github/pull_request_template.md (hunk 2 above)
Comment (baiyli) [reply .github/pull_request_template.md:12]: All github users within our github team are collaborators

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## For Requester
+
+1. Description
+    - [ ] Check the PR title and description for clarity. It should describe the changes made and the reason behind them.
+    - [ ] Ensure that the PR follows the contribution guidelines, if applicable.
+2. Security requirements
+    - [ ] Ensure that a Pull Request (PR) does not expose passwords and other sensitive information by using git-secrets and upload relevant evidence: https://github.com/awslabs/git-secrets
+    - [ ] Ensure commit has GitHub Commit Signature
+3. Automatic checks in GitHub Workflow
```
[the remark this answers is not in the document]
Comment (baiyli) [reply .github/pull_request_template.md:11]: Also remove this part for secret check?
>     - [ ] Ensure that a Pull Request (PR) does not expose passwords and other sensitive information by using git-secrets and upload relevant evidence: https://github.com/awslabs/git-secrets

**Review by jswudi — APPROVED** [review]

**Pull request merged**